ui/components/providers/workflow/forms/update-via-service-account-key-form.tsx

```diff
@@ -0,0 +1,169 @@
+
+  // Handler for back button
+  const handleBackStep = () => {
+    const currentParams = new URLSearchParams(window.location.search);
+    currentParams.delete("via");
+    router.push(`?${currentParams.toString()}`);
+  };
+
+  const providerType = searchParams.type;
+  const providerId = searchParams.id;
+  const providerSecretId = searchParams.secretId || "";
+
+  const formSchema = addCredentialsServiceAccountFormSchema(providerType);
+  type FormSchemaType = z.infer<typeof formSchema>;
+
+  const form = useForm<FormSchemaType>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      providerId,
+      providerType,
+      ...(providerType === "gcp"
+        ? {
+            service_account_key: "",
+            secretName: "",
+          }
+        : {}),
+    },
+  });
+
+  const isLoading = form.formState.isSubmitting;
+
+  const onSubmitClient = async (values: FormSchemaType) => {
+    if (!providerSecretId) {
+      toast({
+        variant: "destructive",
+        title: "Missing Secret ID",
+        description: "Cannot update credentials without a valid secret ID.",
+      });
+      return;
+    }
+
+    const formData = new FormData();
+
+    Object.entries(values).forEach(([key, value]) => {
+      if (value !== undefined && value !== "") {
+        formData.append(key, String(value));
+      }
+    });
+
+    try {
+      const data = await updateCredentialsProvider(providerSecretId, formData);
+      if (data?.errors && data.errors.length > 0) {
+        data.errors.forEach((error: ApiError) => {
+          const errorMessage = error.detail;
+
+          switch (error.source.pointer) {
+            case "/data/attributes/secret/service_account_key":
+              form.setError("service_account_key" as keyof FormSchemaType, {
+                type: "server",
+                message: errorMessage,
+              });
+              break;
+            default:
+              toast({
+                variant: "destructive",
+                title: "Oops! Something went wrong",
+                description: errorMessage,
+              });
+          }
+        });
+      } else {
+        router.push(
+          `/providers/test-connection?type=${providerType}&id=${providerId}&updated=true`,
+        );
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Error during submission:", error);
+      toast({
+        variant: "destructive",
+        title: "Submission failed",
+        description: "An error occurred while processing your request.",
+      });
+    }
+  };
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmitClient)}
+        className="flex flex-col space-y-4"
+      >
+        <input type="hidden" name="providerId" value={providerId} />
+        <input type="hidden" name="providerType" value={providerType} />
+
+        <ProviderTitleDocs providerType={providerType as ProviderType} />
```

Comment:
```suggestion
        <ProviderTitleDocs providerType={providerType} />
```
We don't need this cast